autoload/lsp/diag.vim

```diff
@@ -131,10 +131,17 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
 
     try
       if opt.lspOptions.highlightDiagInline
+	var end_col: number = util.GetLineByteFromPos(bnr, diag.range.end) + 1
+	var end_lnum: number = diag.range.end.line + 1
+	var end_text: string = end_lnum->getline()
+	var end_len: number = end_text->len()
+	if end_col > end_len + 1
+	  end_col = end_len + 1
+	endif
```

Comment:
I think that introducing this code would mean we could remove the catch `E964`, meaning that either we adjust the range (as you do here), well knowing the diagnostics are expired, as new will arrive shortly, or we simply ignore it with the catch.

I still lean towards using try-catch, as in either case the diagnostics that are being rendered are invalid/expired.

Reply:
no, i think for now we needed both, that safeguard code is for one of valid but incorrect cases, but did not know or not sure all invalid cases, so try-catch still required for now.

Reply:
My point is that your new code in only needed for diagnostics that are older than the current document, and new diagnostics will arrive shortly. So I think it is a bad idea to complicate the code for something that can be handled by a `try-catch`.

Since the client speaks utf-32 positions things like:

    int /*😭😭😭😭😭😭😭😭😭😭*/a=b;

Will at worst case result in one of the `😭` being highlighted instead of `b`.

Reply:
you may do not understand the different between 'invalid' with 'incorrect'?
some cases it is incorrect but was valid which would Not try-catch-ed by try-catch.